### PR TITLE
ADCM-3188 add audit for host and cluster update endpoints

### PR DESCRIPTION
**Pull request by SofiaTal**

### CODE REVIEW

File: python/audit/tests/test_cluster.py [hunk 1]
```diff
@@ -132,7 +136,7 @@ def check_log(  # pylint: disable=too-many-arguments
         self.assertEqual(log.operation_result, operation_result)
         self.assertIsInstance(log.operation_time, datetime)
         self.assertEqual(log.user.pk, user.pk)
-        self.assertEqual(log.object_changes, {})
+        self.assertEqual(log.object_changes, object_changes)
```
Comment (winmasta) [inline python/audit/tests/test_cluster.py:139]:
```suggestion
        self.assertDictEqual(log.object_changes, object_changes)
```

File: python/audit/tests/test_cluster.py [hunk 2]
```diff
@@ -487,10 +495,20 @@ def test_update(self):
...
             log=log,
             obj=self.cluster,
-            obj_name=self.cluster.name,
+            obj_name=self.test_cluster_name,
             obj_type=AuditObjectType.Cluster,
             operation_name="Cluster updated",
             operation_type=AuditLogOperationType.Update,
+            object_changes={
+                "current": {
+                    "description": "Blablabla",
```
Comment (winmasta) [inline python/audit/tests/test_cluster.py:504]: `"Blablabla"` duplicated from 488 line, it's better to put it in variable

File: python/audit/tests/test_cluster.py [hunk 3]
```diff
@@ -763,6 +781,69 @@ def test_update_host_config(self):
             operation_type=AuditLogOperationType.Update,
         )
 
+    def test_update_host(self):
+        self.host.cluster = self.cluster
+        self.host.save()
```
Comment (winmasta) [inline python/audit/tests/test_cluster.py:786]: `updated_fields` can be added

File: python/audit/tests/test_cluster.py [hunk 4]
```diff
@@ -763,6 +781,69 @@ def test_update_host_config(self):
...
+            operation_result=AuditLogOperationResult.Fail,
+        )
+        self.host.maintenance_mode = MaintenanceModeType.Off
+        self.host.save()
+        self.client.patch(
+            path=reverse(
+                "host-details",
+                kwargs={"cluster_id": self.cluster.pk, "host_id": self.host.pk},
+            ),
+            data={"description": "Such wow new description", "maintenance_mode": "on"},
```
Comment (winmasta) [inline python/audit/tests/test_cluster.py:830]: Duplicated dict

File: python/audit/utils.py [hunk 5]
```diff
@@ -162,12 +168,20 @@ def _get_obj_changes_data(view: ModelViewSet) -> tuple[dict | None, Model | None
...
             serializer_class = PolicyAuditSerializer
             model = Policy
-
-        if serializer_class:
-            current_obj = model.objects.filter(pk=view.kwargs["pk"]).first()
-            prev_data = serializer_class(model.objects.filter(pk=view.kwargs["pk"]).first()).data
-            if current_obj:
-                prev_data = serializer_class(current_obj).data
+    else:
+        if view.__class__.__name__ == "ClusterDetail" and view.request.method in ["PATCH", "PUT"]:
```
Comment (winmasta) [inline python/audit/utils.py:172]: `in {...}` is faster

**Comment by SofiaTal** [comment]
bender build

**Comment by BenderArenadata** [comment]
Hey, @SofiaTal! Build has been launched!

**Review by winmasta — APPROVED** [review]

**Comment by SofiaTal** [comment]
bender build

**Comment by BenderArenadata** [comment]
Hey, @SofiaTal! Build has been launched!

**Pull request merged**